client/x/evmstaking/keeper/withdrawal_queue.go

```diff
@@ -55,10 +53,9 @@ func (k Keeper) DequeueEligibleWithdrawals(ctx context.Context) (withdrawals ety
 	return withdrawals, nil
 }
 
-func (k Keeper) PeekEligibleWithdrawals(ctx context.Context) (withdrawals etypes.Withdrawals, err error) {
-	maxDequeue, err := k.MaxWithdrawalPerBlock(ctx)
-	if err != nil {
-		return nil, err
+func (k Keeper) PeekEligibleWithdrawals(ctx context.Context, maxPeek int) (withdrawals etypes.Withdrawals, err error) {
+	if maxPeek <= 0 {
```

Comment:
No need to check for negative? Or do we expect `maxPeek` to be a negative value at some point? This should probably be an invariance check (crisis module) for each block.